hijack/permissions.py

```diff
@@ -10,10 +13,14 @@ def superusers_and_staff(*, hijacker, hijacked):
     A superuser may hijack any other user.
     A staff member may hijack any user, except another staff member or superuser.
     """
-    if not hijacked.is_active:
-        return False
+    try:
+        if not hijacked.is_active:
+            return False
 
-    if hijacker.is_superuser:
-        return True
+        if hijacker.is_superuser:
+            return True
 
-    return hijacker.is_staff and not (hijacked.is_staff or hijacked.is_superuser)
+        return hijacker.is_staff and not (hijacked.is_staff or hijacked.is_superuser)
```

Comment:
This part shouldn't be wrapped, since the error is only limited to `hijacked` not `hijacker`.

Reply:
That's right, but it comes with one side-effect:
If we put
``` python
if hijacker.is_superuser:
        return True
```
before the `try`-Block, we lose the check whether `hijacked.is_active` evaluates to True at all. **Currently, this check happens before the `is_superuser` check.**

In order to retain this behavior AND keep the `is_superuser` check outside of the try-Block, it would be necessary to use two try-Blocks:
``` python
def superusers_and_staff(*, hijacker, hijacked):
    """
    Superusers and staff members may hijack other users.

    A superuser may hijack any other user.
    A staff member may hijack any user, except another staff member or superuser.
    """
    try:
        if not hijacked.is_active:
            return False
    except AttributeError:
        return False

    if hijacker.is_superuser:
        return True

    try:
        return hijacker.is_staff and not (hijacked.is_staff or hijacked.is_superuser)

    except AttributeError:
        return False

```

Do you prefer the version with the two try-blocks?

Reply:
Fair enough, I've undone the changes and squashed the commits :-)